doc/build-unix.md

```diff
@@ -277,22 +277,15 @@ A list of additional configure flags can be displayed with:
 
 Setup and Build Example: Arch Linux
 -----------------------------------
-This example lists the steps necessary to setup and build a command line only, non-wallet distribution of the latest changes on Arch Linux:
+This example lists the steps necessary to setup and build a command line only distribution of the latest changes on Arch Linux:
 
-    pacman -S git base-devel boost libevent python
+    pacman --sync --needed autoconf automake boost gcc git libevent libtool make pkgconf python sqlite
     git clone https://github.com/bitcoin/bitcoin.git
     cd bitcoin/
     ./autogen.sh
-    ./configure --disable-wallet --without-gui --without-miniupnpc
+    ./configure
     make check
-
-Note:
-Enabling wallet support requires either compiling against a Berkeley DB newer than 4.8 (package `db`) using `--with-incompatible-bdb`,
```

Comment:
I don't think we're there yet that we should remove instructions for the legacy wallet. I think it makes sense to keep this but explicitly mention that it's needed for the legacy wallet only.

Reply:
@laanwj I agree, but I'm not sure if it belongs to Arch section. This note might help someone build Bitcoin Core with incompatible legacy DB, which is still problematic

There is a separate section which explains how to include compatible `bdb`, and it seems to be the recommended approach: https://github.com/bitcoin/bitcoin/blob/master/doc/build-unix.md#berkeley-db

If the concern is that people might miss this section, maybe it worth adding a link to it instead of repeating similar recipes in every distribution-specific section

Reply:
Yes, agree that adding a link to that section is a better idea. Makes at least sense to mention it. A lot of people will want a `bitcoind` that is compatible with old wallets as well, could be surprised if they simply follow these instructions and it doesn't work.

> This note might help someone build Bitcoin Core with incompatible legacy DB, which is still problematic

Sure, yes. As someone who always builds `bitcoind` from source `--with-incompatible-bdb` is something I automatically type :smile: But from the perspective of someone who uses the guide for the first time it's not a good recommendation.

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/25414/commits/d57a0ed734e4b98b42f13fdf6c05d99434139dc0